pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -444,6 +445,65 @@ func getAddressByName(client *gophercloud.ServiceClient, name types.NodeName) (s
 	return addrs[0].Address, nil
 }
 
+// getInterfacesByName returns the node interfaces of the specified instance.
+func getInterfacesByName(client *gophercloud.ServiceClient, name types.NodeName) ([]attachinterfaces.Interface, error) {
+	var interfaces []attachinterfaces.Interface
+
+	// get server
+	srv, err := getServerByName(client, name)
+	if err != nil {
+		if err == ErrNotFound {
+			return interfaces, cloudprovider.InstanceNotFound
+		}
+		return interfaces, err
+	}
+
+	// get interfaces of server
+	serviceID := srv.ID
+	pager := attachinterfaces.List(client, serviceID)
+
+	interfaceList := make([]attachinterfaces.Interface, 0, 1)
+
+	err = pager.EachPage(func(page pagination.Page) (bool, error) {
+		s, err := attachinterfaces.ExtractInterfaces(page)
+		if err != nil {
+			return false, err
+		}
+		interfaceList = append(interfaceList, s...)
+		return true, nil
+	})
+	if err != nil {
+		return interfaces, err
+	}
+
+	// return ErrNotFound when server contains no interface
+	if len(interfaceList) == 0 {
+		return interfaces, ErrNotFound
+	}
+	return interfaces, nil
+}
+
+// NodeFixedIPToPorts returns the node fixedIPs and ports of the specified instance.
+func getFixedIPToPortsByName(client *gophercloud.ServiceClient, name types.NodeName) (map[string]string, error) {
```

Comment:
This function combines a server lookup, and some arbitrary logic.  It will be easier to test/reuse if it just does one of those two things, and we already have the server lookup function.   So make this just do the arbitrary logic - ie something like `findPortWithFixedIP([]attachinterfaces.Interface, addr string) (string, error)` - or else just drop the function and inline this logic into the caller (since we only want it in one place afaics).